electrumx/server/session/session.py

```diff
@@ -0,0 +1,573 @@
+            'server.ping': self.ping,
+            'server.version': self.server_version,
+        }
+
+        if ptuple >= (1, 4, 2):
+            handlers['blockchain.scripthash.unsubscribe'] = self.scripthash_unsubscribe
+
+        self.request_handlers = handlers
+
+    async def block_header(self, height, cp_height=0):
+        '''Return a raw block header as a hexadecimal string, or as a
+        dictionary with a merkle proof.'''
+        height = non_negative_integer(height)
+        cp_height = non_negative_integer(cp_height)
+        raw_header_hex = (await self.session_mgr.raw_header(height)).hex()
+        self.bump_cost(1.25 - (cp_height == 0))
+        if cp_height == 0:
+            return raw_header_hex
+        result = {'header': raw_header_hex}
+        result.update(await self._merkle_proof(cp_height, height))
+        return result
+
+    async def block_headers(self, start_height, count, cp_height=0):
+        '''Return count concatenated block headers as hex for the main chain;
+        starting at start_height.
+
+        start_height and count must be non-negative integers.  At most
+        MAX_CHUNK_SIZE headers will be returned.
+        '''
+        start_height = non_negative_integer(start_height)
+        count = non_negative_integer(count)
+        cp_height = non_negative_integer(cp_height)
+        cost = count / 50
+
+        max_size = self.MAX_CHUNK_SIZE
+        count = min(count, max_size)
+        headers, count = await self.db.read_headers(start_height, count)
+        result = {'hex': headers.hex(), 'count': count, 'max': max_size}
+        if count and cp_height:
+            cost += 1.0
+            last_height = start_height + count - 1
+            result.update(await self._merkle_proof(cp_height, last_height))
+        self.bump_cost(cost)
+        return result
+
+    async def estimatefee(self, number, mode=None):
+        '''The estimated transaction fee per kilobyte to be paid for a
+        transaction to be included within a certain number of blocks.
+
+        number: the number of blocks
+        mode: CONSERVATIVE or ECONOMICAL estimation mode
+        '''
+        number = non_negative_integer(number)
+        self.bump_cost(2.0)
+        if mode:
+            return await self.daemon_request('estimatefee', number, mode)
+        else:
+            return await self.daemon_request('estimatefee', number)
+
+    async def headers_subscribe(self):
+        '''Subscribe to get raw headers of new blocks.'''
+        self.subscribe_headers = True
+        self.bump_cost(0.25)
+        return await self.subscribe_headers_result()
+
+    async def staking_get_info(self):
+        '''The general information about staking and its parameters'''
+        self.bump_cost(0.2)
+        return await self.daemon_request('getstakinginfo')
+
+    async def stake_get_info(self, hex_hash):
+        '''The general information about staking and its parameters'''
```

Comment:
This comment is copied from the method above, and does not describe the `stake_get_info`